docs/DEBUGGING.md

```diff
@@ -7,13 +7,142 @@
+- `verbose`: Produce verbose log output.
+- `environment`: Environment variables to apply to the ESP-IDF Debug Adapter. It will replace global environment variables and environment variables used by the extension.
+
+```json
+"environment": {
+  "VAR": "Value"
+}
+```
+
+- `imageAndSymbols`:
+
+```json
+"imageAndSymbols": {
+  "symbolFileName": "If specified, a symbol file to load at the given (optional) offset",
+  "symbolOffset": "If symbolFileName is specified, the offset used to load",
+  "imageFileName": "If specified, an image file to load at the given (optional) offset",
+  "imageOffset": "If imageFileName is specified, the offset used to load"
+}
+```
+
+- `target`: Configuration for target to be attached.
+
+```json
+"target": {
+  "type": "The kind of target debugging to do. This is passed to -target-select (defaults to remote)",
+  "host": "Target host to connect to (defaults to 'localhost', ignored if parameters is set)",
+  "port": "Target port to connect to (defaults to value captured by serverPortRegExp, ignored if parameters is set)",
+  "parameters": "Target parameters for the type of target. Normally something like localhost:12345. (defaults to `${host}:${port}`)"
+}
+```
+
+Other arguments please review this extension's package.json `gdbtarget` debugger contribution.
+
+## Use Microsoft C/C++ Extension to Debug
+
+The user can also use [Microsoft C/C++ Extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools) to debug, the community recommend this launch.json configuration:
+
+```JSON
+{
+  "configurations": [
+    {
+      "name": "GDB",
+      "type": "cppdbg",
+      "request": "launch",
+      "MIMode": "gdb",
+      "miDebuggerPath": "${command:espIdf.getXtensaGdb}",
+      "program": "${workspaceFolder}/build/${command:espIdf.getProjectName}.elf",
+      "windows": {
+        "program": "${workspaceFolder}\\build\\${command:espIdf.getProjectName}.elf"
+      },
+      "cwd": "${workspaceFolder}",
+      "environment": [{ "name": "PATH", "value": "${config:idf.customExtraPaths}" }],
+      "setupCommands": [
+        { "text": "target remote :3333" },
+        { "text": "set remote hardware-watchpoint-limit 2"},
+        { "text": "mon reset halt" },
+        { "text": "maintenance flush register-cache" }
+        { "text": "thb app_main" },
+      ],
+      "externalConsole": false,
+      "logging": {
+        "engineLogging": true
+      }
+    }
+  ]
+}
+```
+
+# Using NativeDebug
+
+The user can also try using the [Native Debug](https://marketplace.visualstudio.com/items?itemName=webfreak.debug) extension with this example launch.json configuration:
+
+```JSON
+{
+  "configurations": [
+    {
+      "type": "gdb",
+      "request": "attach",
+      "name": "NativeDebug",
+      "target": "extended-remote :3333",
+      "executable": "${workspaceFolder}/build/${command:espIdf.getProjectName}.elf",
+      "gdbpath": "${command:espIdf.getXtensaGdb}",
```

Comment:
Do we have `getRiscvGdb` function? Again RiscV example would be good to have.

Reply:
Although is called `espIdf.getXtensaGdb ` it does the same for risc v target.

Reply:
`${command:espIdf.getXtensaGdb}` was updated to `${command:espIdf.getToolchainGdb}`